worker/host.go

```diff
@@ -177,15 +199,23 @@ func (h *hostClient) FundAccount(ctx context.Context, desired types.Currency, re
 		}
 
 		// fund the account
-		if err := h.client.FundAccount(ctx, rev, h.hk, h.siamuxAddr, deposit, h.acc.ID(), pt.HostPriceTable, h.renterKey); err != nil {
+		if err := h.rhp3.FundAccount(ctx, rev, h.hk, h.siamuxAddr, deposit, h.acc.ID(), pt.HostPriceTable, h.renterKey); err != nil {
 			if rhp3.IsBalanceMaxExceeded(err) {
 				h.acc.ScheduleSync()
 			}
 			return types.ZeroCurrency, fmt.Errorf("failed to fund account with %v; %w", deposit, err)
 		}
 
 		// record the spend
-		h.contractSpendingRecorder.Record(*rev, api.ContractSpending{FundAccount: deposit.Add(cost)})
+		h.csr.Record(rev.ParentID, func(csr *api.ContractSpendingRecord) {
+			csr.ContractSpending = csr.ContractSpending.Add(api.ContractSpending{FundAccount: deposit.Add(cost)})
+			if rev.RevisionNumber > csr.RevisionNumber {
```

Comment:
This seems like unnecessary duplicate code considering it is going to be the same for every contract spending. I'd rather have `RecordV1` and `RecordV2` if that was the issue rather than this interface.